### ISSUE TITLE
in_channels=1, why sum?

### ISSUE THREAD

**Issue by claverru** [qubvel/segmentation_models.pytorch#363]
https://github.com/qubvel/segmentation_models.pytorch/blob/33dc95082ea76fd83af2a8ebd7a4ecbd549aecfa/segmentation_models_pytorch/encoders/_utils.py#L22

First of all, congratulations on your great library, I found it very useful for different projects. I was wondering if there's any reason behind using *sum* as the aggregation to transform a 3 channels convolution input to a 1 channel one. Isn't the *mean* more intuitive?

The point is that I don't know how to normalize the inputs when using imagenet pretrained weights. Should I also sum the per-channel mean and std?:

IMAGENET_MEAN = [0.485, 0.456, 0.406] --> sum([0.485, 0.456, 0.406])
IMAGENET_STD = [0.229, 0.224, 0.225] --> sum([0.229, 0.224, 0.225])

Thank you very much!

**Comment by qubvel** [comment]
#341

**Comment by JulienMaille** [comment]
You should follow what is done in the [quickstart.rst](https://github.com/qubvel/segmentation_models.pytorch/blob/master/docs/quickstart.rst) the pre-processing will take care of mean/std normalisation. 
https://github.com/qubvel/segmentation_models.pytorch/blob/176f2777cbfc4101abf9b10d1339c7a3e93f06f2/segmentation_models_pytorch/encoders/_preprocessing.py#L4-L23

**Comment by claverru** [comment]
@JulienMaille Thank you for your post, but I dont think it answers my question. 

What I want to know is, if I have only one channel as input but imagenet mean and std are prepared for three channels, how should I aggregate those to apply them say, in a function like you pasted. Basically how should I pass mean and std args to that function. 

I'm using albumentations as preprocessing tool but their normalization function should be similar to the one you pointed there. The question remains the same: how do I aggregate mean and std from shape (3, ) to shape (1, ).

Thank you!

**Comment by JulienMaille** [comment]
I just noticed that on my fork I modified that bit to:
```python
    in_dim = x.shape[2]
    if mean is not None:
        mean = np.array(mean)
        x = x - mean[0:in_dim]

    if std is not None:
        std = np.array(std)
        x = x / std[0:in_dim]
```

**Comment by JulienMaille** [comment]
Well I had to read it twice, but I think qubvel link to #341 is misleading. Convolution weights needs to be summed (not averaged) as he explained. But when it comes to preprocessing, either take the average or the `[:in_channels]`
Someone corrects me if I'm wrong.

**Comment by claverru** [comment]
Well, he answered part of my question, the one that is announced in the title. For the other part you helped me already.
